lib/mulang/ruby/ast_processor.rb

```diff
@@ -189,6 +189,37 @@ def on_lvasgn(node)
       ms :Assignment, id, process(value)
     end
 
+    def on_op_asgn(node)
+      assignee, message, value = *node
+
+      if assignee.type == :send
+        property_assignment assignee, message, value
+      else
+        var_assignment assignee, message, value
+      end
+    end
+
+    def var_assignment(assignee, message, value)
+      id = assignee.to_a.first
+      ms :Assignment, id, simple_send(ms(:Reference, id), message, [process(value)])
+    end
+
+    def property_assignment(assignee, message, value)
+      receiver, messsage, args = *assignee
+      id = process(receiver)
+      simple_send id, "#{messsage}=".to_sym, [process(args), simple_send(simple_send(id, messsage, [process(args)].compact), message, [process(value)])].compact
```

Comment:
this line desperately needs a refator but I can't think of one. @flbulgarelli ?

Reply:
refator 
![bitmoji](https://render.bitstrips.com/v2/cpanel/8f94e6d0-4aac-4c50-a609-18e87cd7cc33-ffe5b08d-c271-4b7d-b267-708d3e0b004c-v1.png?transparent=1&palette=1&width=246)

Reply:
Thanks for so much @aguspina